PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java

```diff
@@ -26,192 +31,210 @@
+        gridData = new GridData();
+        gridData.horizontalAlignment = SWT.FILL;
+        gridData.grabExcessHorizontalSpace = true;
+        storageNameField.setLayoutData(gridData);
+        storageNameField.setToolTipText("The default storage account of the HDInsight cluster, which can be found from HDInsight cluster properties of Azure portal.\n\n Press the F1 key or click the '?'(Help) button to get more details");
+
+        Label storageKeyLabel = new Label(container, SWT.LEFT);
+        storageKeyLabel.setText("Storage Key:");
+        gridData = new GridData();
+        gridData.horizontalAlignment = SWT.RIGHT;
+        storageKeyLabel.setLayoutData(gridData);
+        storageKeyField = new Text(container, SWT.BORDER);
+        gridData = new GridData();
+        gridData.horizontalAlignment = SWT.FILL;
+        gridData.grabExcessHorizontalSpace = true;
+        storageKeyField.setLayoutData(gridData);
+        storageKeyField.setToolTipText("The storage key of the default storage account, which can be found from HDInsight cluster storage accounts of Azure portal.\n\n Press the F1 key or click the '?'(Help) button to get more details.");
+
+        Label userNameLabel = new Label(container, SWT.LEFT);
+        userNameLabel.setText("User Name:");
+        gridData = new GridData();
+        gridData.horizontalAlignment = SWT.RIGHT;
+        userNameLabel.setLayoutData(gridData);
+        userNameField = new Text(container, SWT.BORDER);
+        gridData = new GridData();
+        gridData.horizontalAlignment = SWT.FILL;
+        gridData.grabExcessHorizontalSpace = true;
+        userNameField.setLayoutData(gridData);
+        userNameField.setToolTipText("The user name of the HDInsight cluster.\n\n Press the F1 key or click the '?'(Help) button to get more details.");
+        
+        Label passwordLabel = new Label(container, SWT.LEFT);
+        passwordLabel.setText("Password:");
+        gridData = new GridData();
+        gridData.horizontalAlignment = SWT.RIGHT;
+        passwordLabel.setLayoutData(gridData);
+        passwordField = new Text(container, SWT.PASSWORD | SWT.BORDER);
+        gridData = new GridData();
+        gridData.horizontalAlignment = SWT.FILL;
+        gridData.grabExcessHorizontalSpace = true;
+        passwordField.setLayoutData(gridData);
+        passwordField.setToolTipText("The password of the HDInsight cluster user provided.\n\n Press the F1 key or click the '?'(Help) button to get more details.");
+        
+        container.addHelpListener(new HelpListener() {
+            @Override public void helpRequested(HelpEvent e) {
+                try {
+                    IWebBrowser webBrowser = PlatformUI.getWorkbench().getBrowserSupport().getExternalBrowser();
+                    if (webBrowser != null)
+                        webBrowser.openURL(new URL("https://go.microsoft.com/fwlink/?linkid=866472"));
+                } catch (Exception e1) {
```

Comment:
Log the exception or `Exception ignored`?

Reply:
added logger in the new commit.

Reply:
You can choose Long's second option here